Dockerfile

```diff
@@ -1,44 +1,54 @@
-FROM golang:1.15-alpine as builder
-
-ARG TARGETPLATFORM
+FROM --platform=$BUILDPLATFORM golang:1.15-alpine as gomod
 
 WORKDIR /workspace
 
-RUN apk add --no-cache ca-certificates curl
-
-RUN kubectl_ver=1.20.2 && \
-arch=${TARGETPLATFORM:-linux/amd64} && \
-if [ "$TARGETPLATFORM" == "linux/arm/v7" ]; then arch="linux/arm"; fi && \
-curl -sL https://storage.googleapis.com/kubernetes-release/release/v${kubectl_ver}/bin/${arch}/kubectl \
--o /usr/local/bin/kubectl && chmod +x /usr/local/bin/kubectl
-
-RUN kubectl version --client=true
-
 # copy api submodule
 COPY api/ api/
 
 # copy modules manifests
-COPY go.mod go.mod
-COPY go.sum go.sum
+COPY go.mod go.sum ./
 
 # cache modules
 RUN go mod download
 
-# copy source code
-COPY main.go main.go
-COPY controllers/ controllers/
-COPY internal/ internal/
+# ------------------------------------------------------------------------------
```

Comment:
Instead of putting a divider here, it is better to write a short comment on what comes next. If you need it to "stand out", you can format the doc block a bit, e.g:

```Dockerfile
#
# This stands out, and is where the build part starts
#
FROM --platform=$BUILDPLATFORM golang:1.15-alpine as builder

# While this is a short comment
ARG TARGETPLATFORM

# ...omitted for brevity

#
# This is the actual container image
#
FROM --platform=$TARGETPLATFORM alpine:3.13

# ...omitted for brevity
````

Reply:
I really think dividers help having a better vision of the different steps.

I can add comments if you want to but I'd like to keep the dividers for clarity.

Reply:
Dividers only help (some*) if you know how to navigate the file, while explanatory comments are useful to anyone reading it.

Given changes to the `Dockerfile` are scarce and we are developing OSS, people looking at it will often not be familiar with what is in the file, and we should thus aim for the latter.

---

\* I _personally_ find that they just create noise